### ISSUE TITLE
LINE_EXISTS suggestions for non-EQ search keys

### ISSUE THREAD

**Issue by GKaschke** [SAP/code-pal-for-abap#608]
Hello,
Are LINE_EXISTS suggestions correct for non-EQ search keys in READ or LOOP?
As far as I know, I can only specify keys in LINE_EXISTS() via EQ
Regards,
Günter

Example: 
   clear r_yes.
    loop at gritab transporting no fields where     ico_towork   = abap_true
                                                and ico_created <> zcl_zhplm_utils=>c_ico_created_complete.
      r_yes = abap_true.
      exit.
    endloop.

**Comment by bjoern-jueliger-sap** [comment]
Yeah, this is a bug (the [Cloud version](https://github.com/SAP/code-pal-for-abap-cloud/pull/52) of this check already doesn't have that anymore).

This implementation of the check [does some analysis for `LOOP`s ](https://github.com/SAP/code-pal-for-abap/blob/58e9f75b14c587ef448d7c454eaed5ff04f10d99/src/checks/y_check_prefer_line_exists.clas.abap#L59) but there is no checking whether the condition is actually valid as a table expression (and hence could be transformed into `line_exists`).

**Comment by Korochenkov** [comment]
> Yeah, this is a bug (the [Cloud version](https://github.com/SAP/code-pal-for-abap-cloud/pull/52) of this check already doesn't have that anymore).
> 
> This implementation of the check [does some analysis for `LOOP`s ](https://github.com/SAP/code-pal-for-abap/blob/58e9f75b14c587ef448d7c454eaed5ff04f10d99/src/checks/y_check_prefer_line_exists.clas.abap#L59) but there is no checking whether the condition is actually valid as a table expression (and hence could be transformed into `line_exists`).

Hello,

Today's 2026 year and the same issue still exists. Do you have plans to adjust the check?
